packages/enhanced/src/runtime/ModuleInfoRuntimeModule.ts

```diff
@@ -21,11 +21,20 @@ export class ModuleInfoRuntimeModule extends RuntimeModule {
   override generate(): string {
     return Template.asString([
       `${RuntimeGlobals.require}.federation = {`,
-      Template.indent([`cache: {},`, `remotes: {},`, `moduleInfo: {},`]),
+      Template.indent([`cache: {},`, `remotes: {},`, `moduleInfo: {}, `]),
       `};`,
       Template.indent([
         '// backward compatible global proxy',
         `let oldScope = globalThis.__remote_scope__ || {};`,
+             //TODO: move this elsewhere
+        `if(${RuntimeGlobals.runtimeId} === 'webpack-runtime' && globalThis.__remote_scope__) {`,
+          // during lazy compilations, webpack will push updated runtime modules ito webpack runtime
```

Comment:
This actually enables us to now dispose and reinitialize containers. In future iteration we should be able to support real hot module reloading on server side, in production environment.

Reply:
Even now, I should be able to perform simple and consistent reloading in node, all I need to do is clear require cache of the webpack runtime. It will reinitialize more gracefully than before